src/UI/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -40,7 +40,9 @@ public class InputModel
         public virtual Task<IActionResult> OnPostConfirmationAsync(string returnUrl = null) => throw new NotImplementedException();
     }
 
-    internal class ExternalLoginModel<TUser> : ExternalLoginModel where TUser : IdentityUser, new()
+    internal class ExternalLoginModel<TUser, TKey> : ExternalLoginModel
```

Comment:
So we actually need to get rid of the IdentityUser constraint entirely...and we certainly don't want to bring in the TKey either...

I believe the new TUser calls are what's tying you IdentityUser since its using the UserName/Email properties to initialize.

I think if you introduce a UI service like `TUser IUIUserInstanceFactory<TUser>.Create(string userName, string email`, and we can add some logic to AddDefaultUI to automatically register a concrete `UserInstanceFactory<TUser>` that does what we do today whenever TUser is an IdentityUser, otherwise we don't, and the app will need to register the factory

Reply:
I would not introduce a new abstraction. Simply force TUser to be a class and have a parameterless constructor

Reply:
I would suggest to start by replacing any access to user data with method calls from user manager and then apply the minimum required set of constraints

Reply:
Yeah unfortunately that won't work, CreateAsync validates the user which requires a user name to be set at a minimum, and potentially a non null email if that was required as well, otherwise all usermanager operations will be rejected.  The idea is that it was up to the callee to create TUser however they wish and then call CreateAsync when ready...  Its an abstraction that isn't needed for IdentityCore, but its now a new abstraction that is needed for the UI :(